LibrePCB-2.0.1 does not support OpenCascade 8_0_0

##### VERSION / OS / ENVIRONMENT

LibrePCB Version: 2.0.1
Git Revision:
Qt Version: 6.10.2
Operating System: FreeBSD 16-Current
Note: this is independent of the OS, it only depends on the release of OCCT.

##### SUMMARY
Compilation failure with OCCT 8_0_0.

##### STEPS TO REPRODUCE
Compiling LibrePCB 2.0.1 fails with theses errors:
```
[  8%] Building CXX object libs/parseagle/CMakeFiles/parseagle.dir/parseagle/schematic/net.cpp.o
cd /wrkdirs/usr/ports/cad/librepcb/work/.build/libs/parseagle && /usr/local/libexec/ccache/c++ -DGL_SILENCE_DEPRECATION -DHAVE_DRACO -DHAVE_FFMPEG -DHAVE_FREEIMAGE -DHAVE_FREETYPE -DHAVE_GLES2_EXT -DHAVE_OPENGL_EXT -DHAVE_RAPIDJSON -DHAVE_TBB -DHAVE_TK -DHAVE_VTK -DHAVE_XLIB -DOCCT_NO_DEPRECATED -DOCC_CONVERT_SIGNALS -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x060200 -DQT_NO_DEBUG -DQT_NO_DEPRECATED_WARNINGS -DQT_XML_LIB -DVTK_OPENGL2_BACKEND -I/wrkdirs/usr/ports/cad/librepcb/work/.build/libs/parseagle -I/wrkdirs/usr/ports/cad/librepcb/work/librepcb-2.0.1/libs/parseagle -isystem /usr/local/include/qt6/QtCore -isystem /usr/local/include/qt6 -isystem /usr/local/lib/qt6/mkspecs/freebsd-clang -isystem /usr/local/include/qt6/QtXml -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include  -DNDEBUG -std=c++17 -pthread -MD -MT libs/parseagle/CMakeFiles/parseagle.dir/parseagle/schematic/net.cpp.o -MF CMakeFiles/parseagle.dir/parseagle/schematic/net.cpp.o.d -o CMakeFiles/parseagle.dir/parseagle/schematic/net.cpp.o -c /wrkdirs/usr/ports/cad/librepcb/work/librepcb-2.0.1/libs/parseagle/parseagle/schematic/net.cpp
/wrkdirs/usr/ports/cad/librepcb/work/librepcb-2.0.1/libs/librepcb/core/3d/occmodel.cpp:576:7: error: unknown type name 'TopTools_ListOfShape'
  576 |       TopTools_ListOfShape boardFaces;
      |       ^
/wrkdirs/usr/ports/cad/librepcb/work/librepcb-2.0.1/libs/librepcb/core/3d/occmodel.cpp:578:7: error: unknown type name 'TopTools_ListOfShape'
  578 |       TopTools_ListOfShape holeFaces;
      |       ^
2 errors generated.
gmake[2]: *** [libs/librepcb/core/CMakeFiles/librepcb_core.dir/build.make:96: libs/librepcb/core/CMakeFiles/librepcb_core.dir/3d/occmodel.cpp.o] Error 1
```

##### EXPECTED RESULTS
Compilation is OK.

##### ACTUAL RESULTS
Compilation failure.

##### ADDITIONAL INFORMATION
N/A

--- a/libs/librepcb/core/3d/occmodel.cpp
+++ b/libs/librepcb/core/3d/occmodel.cpp
@@ -47,6 +47,7 @@
 #include <Message_Messenger.hxx>
 #include <Message_Printer.hxx>
 #include <Message_SequenceOfPrinters.hxx>
+#include <NCollection_List.hxx>
 #include <Poly_Triangulation.hxx>
 #include <Quantity_Color.hxx>
 #include <Standard_Version.hxx>
@@ -616,9 +617,9 @@ std::unique_ptr<OccModel> OccModel::createBoard(const librepcb::Path& outline,
       // Cutting fails if there are no holes.
       face = pathToFace(outline.cleaned(), Length(0));
     } else {
-      TopTools_ListOfShape boardFaces;
+      NCollection_List<TopoDS_Shape> boardFaces;
       boardFaces.Append(pathToFace(outline.cleaned(), Length(0)));
-      TopTools_ListOfShape holeFaces;
+      NCollection_List<TopoDS_Shape> holeFaces;
       foreach (const Path& hole, holes) {
         holeFaces.Append(pathToFace({hole.cleaned()}, Length(0)));
       }

